src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -94,16 +94,21 @@ async function put_bucket_lifecycle(req) {
         }
         current_rule.status = rule.Status[0];
 
-        if (rule.Prefix) {
-            dbg.error('Rule should not have prefix, it should be filter.prefix', rule);
-            throw new S3Error(S3Error.InvalidArgument);
+        if (rule.Prefix && rule.Filter) {
```

Comment:
Why readding rule.Prefix? 
AWS has deprecated it, and we followed:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_LifecycleRule.html
https://github.com/noobaa/noobaa-core/pull/6849

Reply:
I believe its due to the test that is checking it.

We can decide to skip and open a bug on the tests repo

Reply:
Sound like a plan 🙂 
We should do that.
@nadavMiz Can you revert the lifecycle change, skip that test, open an issue on the test repo, and link it here?
We should pay attention that as Ceph might still support Prefix directly, they might not fix the test (We took care of it in an upgrade script).

Reply:
even though rule.Prefix is deprecated, it is still available by AWS. that is the reason I added it
didn't realize it was removed intentionally. I can remove the commit. if we don't want it

Reply:
I have reverted the changes. this is the the issue on the test repo: https://github.com/ceph/s3-tests/issues/510

I have kept test_lifecycle_transition_set_invalid_date in the pending list because even though it passes, the transition feature is not implemented and test should fail

Reply:
Issue was opened and changes done , @liranmauda